qadence2_expressions/core/constructors.py

```diff
@@ -63,7 +33,8 @@ def promote(x: Expression | complex | float | int) -> Expression:
     """Type cast inputs as value type expressions.
 
     Args:
-        x (Expression | complex | float | int): A valid expression or numerical value. Numerical values are converted into `Value(x)` expressions.
+        x (Expression | complex | float | int): A valid expression or numerical value.
```

Comment:
I guess the bottomline is: `Numeric` type is good as it makes `Expression` more readable than to carry over this long union type everywhere.

Reply:
I'm with @RolandMacDoland that the `Numeric` type is a good addition.

Reply:
The problem with using `Numeric` protocol as before is that there will be the need to implement some exception handlings throughout the `Expression` class. For instance:

```
from qadence2_expressions import Numeric
import torch
isinstance(torch.tensor(1.0, requires_grad=True), Numeric)

>>
Traceback (most recent call last):
  File "/Users/emaschio/Library/Application Support/hatch/env/virtual/qadence2-expressions/WGNTQKyX/qadence2-expressions/lib/python3.10/site-packages/IPython/core/interactiveshell.py", line 3577, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-2-72c52bf21fe0>", line 4, in <module>
    isinstance(torch.tensor(1.0, requires_grad=True), Numeric)
  File "/opt/homebrew/Cellar/python@3.10/3.10.15/Frameworks/Python.framework/Versions/3.10/lib/python3.10/typing.py", line 1506, in __instancecheck__
    if all(hasattr(instance, attr) and
  File "/opt/homebrew/Cellar/python@3.10/3.10.15/Frameworks/Python.framework/Versions/3.10/lib/python3.10/typing.py", line 1506, in <genexpr>
    if all(hasattr(instance, attr) and
RuntimeError: imag is not implemented for tensors with non-complex dtypes.
```

while 

```
from qadence2_expressions import Numeric
import torch
isinstance(torch.tensor(1.0j, requires_grad=True), Numeric)

>> True
```

Reply:
nevermind. Replaced it by `Numeric = Union[complex | float | int]` alias